langserver/handlers/execute_command.go

```diff
@@ -13,12 +13,26 @@ import (
 type executeCommandHandler func(context.Context, commandArgs) (interface{}, error)
 type executeCommandHandlers map[string]executeCommandHandler
 
-var handlers = executeCommandHandlers{
-	"rootmodules": executeCommandRootModulesHandler,
+const langServerPrefix = "terraform-ls."
+
+var commandPrefix string
+var handlers = make(executeCommandHandlers)
```

Comment:
Technically the same server instance/process can serve multiple sessions, each of which may be initialized with a different prefix, wouldn't such shared state lead to race conditions?

Reply:
Can it? Do we support that or are you saying the spec in general does? Once the language server supports multiple workspaces i was going to deprecate this setting

Reply:
After discussion I see the foolishness of my assumption that our languageserver currently only support 1 session. This is not true and this PR uses global memory which will not work out well. I will correct the implementation, fortunately there are no changes to the public API proposed, just internals.